LabManagement/PowerShell/BulkOperations/Az.LabServices.BulkOperations.psm1

```diff
@@ -1438,6 +1438,178 @@ function Remove-AzLabUsersBulk {
+    )
+
+    begin {
+        # This patterns aggregates all the objects in the pipeline before performing an operation
+        # i.e. executes lab creation in parallel instead of sequentially
+        # This trick is to make it work both when the argument is passed as pipeline and as normal arg.
+        # I came up with this. Maybe there is a better way.
+        $aggregateLabs = @()
+    }
+    process {
+        # If passed through pipeline, $labs is a single object, otherwise it is the whole array
+        # It works because PS uses '+' to add objects or arrays to an array
+        $aggregateLabs += $labs
+
+    }
+    end {
+        $init = {            
+        }
+
+        function Send-AzLabsInvitation-Jobs {
+            [CmdletBinding()]
+            param(
+                [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+                [psobject[]]
+                $ConfigObject
+            )
+
+            $block = {
+                param($path)
+
+                Set-StrictMode -Version Latest
+                $ErrorActionPreference = 'Stop'
+
+                Import-Module -Name Az.LabServices
+
+                $input.movenext() | Out-Null
+            
+                $obj = $input.current[0]
+            
+                if ($obj.Invitation -and $obj.Emails) { 
+                    Write-Host "Sending invitation emails for lab: $($obj.LabName)."
+                    $lab = Get-AzLabServicesLab -ResourceGroupName $obj.ResourceGroupName -Name $($obj.LabName)
+                    if (-not $lab -or @($lab).Count -ne 1) { Write-Error "Unable to find lab $($obj.LabName)." }
+
+                    $users = Get-AzLabServicesUser -LabName $lab.Name -ResourceGroupName $obj.ResourceGroupName
+                    if (($users | Measure-Object).Count -eq 0) { Write-Error "The lab doesn't have any users added." }
+
+                    foreach ($user in $users ) {
+                        Write-Verbose "Sending invitation email to user: $($user.Email)."
+                        Send-AzLabServicesUserInvite -ResourceGroupName $obj.ResourceGroupName -LabName $obj.LabName -UserName $user.name -Text $obj.Invitation | Out-Null
+                    }
+                }
+                else {
+                    Write-Error "The invitation and\or user emails are missing for lab in the input .csv file: $($obj.LabName)."
+                }
+            }
+
+            Write-Host "Sending lab invitations to users.  This may take awhile."
+            $jobs = @()
+
+            $ConfigObject | ForEach-Object {
+            Write-Verbose "From config: $_"
+                $jobs += Start-ThreadJob  -InitializationScript $init -ScriptBlock $block -ArgumentList $PSScriptRoot -InputObject $_ -Name ("$($_.ResourceGroupName)+$($_.LabPlanName)+$($_.LabName)") -ThrottleLimit $ThrottleLimit
+            }
+
+            return JobManager -currentJobs $jobs -ResultColumnName "PublishResult" -ConfigObject $ConfigObject
```

Comment:
Probably need to update the column name for the output to be "InvitationSentResult" or something?

Reply:
Yes, another copy\paste mistake...thanks for catching.